string curr_state
string desired_color_detect